products/views.py

```diff
@@ -1 +1,35 @@
-from django.shortcuts import render
\ No newline at end of file
+import json
+
+from django.http      import JsonResponse
+from django.views     import View
+
+from products.models import Product, Image
+
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        if not Product.objects.filter(id=product_id).exists():
+            return JsonResponse({'message' : 'PRODUCT_DOES_NOT_EXISTS'}, status=404)
+
+        product = Product.objects.get(id=product_id)
+
+        images  = Image.objects.filter(product=product.id)
+        image_urls = [image.url for image in images]
+        image_urls.insert(0,product.thumbnail_url)
```

Comment:
```suggestion
        product = Product.objects.get(id=product_id)
        images  = product.image_set.all()
        
        image_urls = [image.url for image in images]
        image_urls.insert(0,product.thumbnail_url)
```
Image class로부터 image를 잘 가져오셨습니다.

그런데 product 객체를 가져온 상태라면 다음과 같이 가져 올 수도 있습니다.